openstef/feature_engineering/missing_values_transformer.py

```diff
@@ -83,17 +94,25 @@ def transform(self, x) -> pd.DataFrame:
 
         x = x[self.non_null_feature_names]
 
-        return self.imputer_.transform(x)
+        transformed = self.imputer_.transform(x)
 
-    def fit_transform(self, x, y=None):
+        return transformed
+
+    def fit_transform(self, x, y=None) -> tuple[pd.DataFrame, Optional[pd.Series]]:
         """Fit the imputer on the input data and transform it.
 
         Returns:
             The data with missing values imputed.
 
         """
         self.fit(x, y)
-        return self.transform(x)
+
+        if y is not None:
+            y = y.loc[self.non_trailing_null_rows]
+
+        x = self.transform(x)
+        x = x.loc[self.non_trailing_null_rows]
```

Comment:
We probably should using self.non_trailing_null_rows and instead just recalculate non_trailing_null_rows.

Mainly because if you set it to a member variable `non_trailing_null_rows` then it will also be baked into model's pickle checkpoint even though it is not used beyond the fitting process.